### PR TITLE
fix: removing #[tokio::main] macro from create_bucket

### PR DESCRIPTION
**Pull request by gabrik**

### CODE REVIEW

**Comment by gabrik** [comment]
@DariusIMP

**Comment by gabrik** [comment]
Given that this also involves runtimes @YuanYuYuan please have a look.

It is a workaround to make it work both with dynamic loading and static linking.

**Review by DariusIMP — APPROVED** [review]
LGTM

File: src/lib.rs [hunk 1]
```diff
@@ -134,22 +140,46 @@ impl Volume for S3Volume {
...
+                config.credentials.to_owned(),
+                config.bucket.to_owned(),
+                self.region.to_owned(),
+                self.endpoint.to_owned(),
+                self.tls_config.to_owned(),
+            )
+            .await,
+        );
+
+        let storage_runtime = tokio::runtime::Builder::new_multi_thread()
```
[the remark this answers is not in the document]
Comment (gabrik) [reply src/lib.rs:154]: I indeed did a mistake it should only be declared once.

File: src/lib.rs [hunk 2]
```diff
@@ -68,6 +68,12 @@ impl Plugin for S3Backend {
     const PLUGIN_LONG_VERSION: &'static str = zenoh_plugin_trait::plugin_long_version!();
 
     fn start(_name: &str, config: &Self::StartArgs) -> ZResult<Self::Instance> {
+        #[cfg(feature = "dynamic_plugin")]
+        let _rt = tokio::runtime::Builder::new_multi_thread()
```
[the remark this answers is not in the document]
Comment (gabrik) [reply src/lib.rs:72]: Mistake again, I was creating a runtime at start but I noticed after that a `storage_runtime` was being crated anyway.

It must have survived the cleanup, I'll fix it.

**Pull request merged**